src/transaction.rs

```diff
@@ -1,42 +1,127 @@
+use crate::alloy_u64_to_ethers;
 use crate::request_shim::{AlloyTransactionRequest, TransactionRequestShim};
+use alloy_primitives::{Address, U256, U64};
+use alloy_sol_types::SolCall;
 use ethers::middleware::SignerMiddleware;
-use ethers::providers::Middleware;
+use ethers::providers::{JsonRpcClient, Middleware, Provider, ProviderError};
 use ethers::signers::Signer;
-use ethers::types::{Eip1559TransactionRequest, TransactionReceipt};
+use ethers::types::transaction::eip2718::TypedTransaction;
+use ethers::types::TransactionReceipt;
 use ethers::utils::hex;
 use tracing::info;
 
-pub struct ExecutableTransaction<M: Middleware, S: Signer> {
-    pub transaction_request: Eip1559TransactionRequest,
-    pub client: SignerMiddleware<M, S>,
+pub struct WriteContractParameters<C: SolCall> {
+    pub call: Option<C>,
+    pub address: Option<Address>,
+    pub gas: Option<U256>,
+    pub gas_price: Option<U256>,
+    pub max_fee_per_gas: Option<U256>,
+    pub max_priority_fee_per_gas: Option<U256>,
+    pub nonce: Option<U256>,
+    pub value: Option<U256>,
 }
 
-impl<M: Middleware, S: Signer> ExecutableTransaction<M, S> {
-    pub async fn from_alloy_transaction_request(
-        transaction_request: AlloyTransactionRequest,
-        client: SignerMiddleware<M, S>,
-    ) -> anyhow::Result<Self> {
-        Ok(Self {
-            transaction_request: transaction_request.to_eip1559(),
-            client,
-        })
+impl<C: SolCall> WriteContractParameters<C> {
+    pub fn default() -> Self {
```

Comment:
if everything is an option then this would be what you get with `#[derive(Default)]` anyway